woocommerce/woocommerce-android
# Auto-sync order editing

Closes: #6730
Closes: #6731


### Description
As discussed [here](p91TBi-91L-p2), when editing an order, we need to commit all local changes to the remote order. We already have a strategy to auto-sync price modifiers when the order is in the Creation flow. With this PR, I wanted to abstract the common functionality from CreateUpdateOrder and create two strategies for syncing orders. The first one, AutoSyncPriceModifier, will be used in the Creation flow and ensure the order will commit changes only when these changes affect price modifiers (what we already have). The second one, AutoSyncOrder, will commit all changes made by the user to the remote order.

### Testing instructions

TC 1
1. run CreationFocusedOrderCreationViewModelTest
2. run EditFocusedOrderCreationViewModelTest
3. run CreateUpdateOrderTests
4. Ensure all tests pass

TC 2
1. Select Orders -> Create order
2. Made some changes affecting the price of the order (add a product or fee)
3. Check that the changes are committed to the remote order
4. Made other changes like changing the order status or the Customer info
5. Check that the changes are NOT committed to the remote order

TC 3
1. Select Orders -> Select an order -> More menu (3 dots) -> Edit
2. Made some changes to the order
3. Check that the changes are committed to the remote order


### Images/gif


#### Creation flow
https://user-images.githubusercontent.com/18119390/176469948-7bcbce10-d202-43c4-9ef3-e85325925378.mp4

#### Edition flow
https://user-images.githubusercontent.com/18119390/176469990-45c3d695-33dc-449b-b0fb-95dc69d8af97.mp4
